### PR TITLE
feat: Adds drill to detail context menu to Pivot Table

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Adds drill-to-detail context menu to Pivot Table chart when right-clicking on the cells. It also improves the context menu positioning for the other supported chart types.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/70410625/186673147-70e66732-7dd3-4399-8e36-385212a5a5be.mov

### TESTING INSTRUCTIONS
Check the video for test instructions.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/plugins/plugin-chart-pivot-table/src/PivotTableChart.tsx [hunk 1]
```diff
@@ -360,6 +362,46 @@ export default function PivotTableChart(props: PivotTableProps) {
     [colSubtotalPosition, rowSubtotalPosition],
   );
 
+  const handleContextMenu = (
```
Comment (kgabryje) [inline superset-frontend/plugins/plugin-chart-pivot-table/src/PivotTableChart.tsx:365]: Can we wrap it in `useCallback`?

**Review by kgabryje — APPROVED** [review]
Looks great!
1 nitpick about `useCallback` and 1 cosmetic question/suggestion - should we use a different cursor to indicate that the cells are clickable? Maybe a "normal" one (pointer would suggest left-click instead of right-click)?

File: superset-frontend/plugins/plugin-chart-pivot-table/src/PivotTableChart.tsx (hunk 1 above)
Comment (michael-s-molina) [reply superset-frontend/plugins/plugin-chart-pivot-table/src/PivotTableChart.tsx]: Done

**Comment by michael-s-molina** [comment]
> Looks great! 1 nitpick about `useCallback` and 1 cosmetic question/suggestion - should we use a different cursor to indicate that the cells are clickable? Maybe a "normal" one (pointer would suggest left-click instead of right-click)?

That's a good question. I'll bring it to the team and if we decide to use a specific cursor then I'll do it in a follow-up because I need to change all plugins.

**Pull request merged**